python_modules/libraries/dagster-databricks/dagster_databricks/ext.py

```diff
@@ -0,0 +1,156 @@
+        submit_args: Optional[Mapping[str, str]] = None,
+    ) -> None:
+        """Run a Databricks job with the EXT protocol.
+
+        Args:
+            task (databricks.sdk.service.jobs.SubmitTask): Specification of the databricks
+                task to run. Environment variables used by dagster-ext will be set under the
+                `spark_env_vars` key of the `new_cluster` field (if there is an existing dictionary
+                here, the EXT environment variables will be merged in). Everything else will be
+                passed unaltered under the `tasks` arg to `WorkspaceClient.jobs.submit`.
+            submit_args (Optional[Mapping[str, str]]): Additional keyword arguments that will be
+                forwarded as-is to `WorkspaceClient.jobs.submit`.
+        """
+        with ext_protocol(
+            context=context,
+            extras=extras,
+            context_injector=context_injector or ExtDbfsContextInjector(client=self.client),
+            message_reader=message_reader or ExtDbfsMessageReader(client=self.client),
+        ) as ext_context:
+            submit_task_dict = task.as_dict()
+            submit_task_dict["new_cluster"]["spark_env_vars"] = {
+                **submit_task_dict["new_cluster"].get("spark_env_vars", {}),
+                **(self.env or {}),
+                **ext_context.get_external_process_env_vars(),
+            }
+            task = jobs.SubmitTask.from_dict(submit_task_dict)
+            run_id = self.client.jobs.submit(
+                tasks=[task],
+                **(submit_args or {}),
+            ).bind()["run_id"]
+
+            while True:
+                run = self.client.jobs.get_run(run_id)
+                if run.state.life_cycle_state in (
+                    jobs.RunLifeCycleState.TERMINATED,
+                    jobs.RunLifeCycleState.SKIPPED,
+                ):
+                    if run.state.result_state == jobs.RunResultState.SUCCESS:
+                        return
+                    else:
+                        raise DagsterExternalExecutionError(
+                            f"Error running Databricks job: {run.state.state_message}"
+                        )
+                elif run.state.life_cycle_state == jobs.RunLifeCycleState.INTERNAL_ERROR:
+                    raise DagsterExternalExecutionError(
+                        f"Error running Databricks job: {run.state.state_message}"
+                    )
+                time.sleep(5)
+
+
+ExtDatabricks = ResourceParam[_ExtDatabricks]
+
+_CONTEXT_FILENAME = "context.json"
+
+
+@contextmanager
+def dbfs_tempdir(dbfs_client: files.DbfsAPI) -> Iterator[str]:
+    dirname = "".join(random.choices(string.ascii_letters, k=30))
+    tempdir = f"/tmp/{dirname}"
+    dbfs_client.mkdirs(tempdir)
+    try:
+        yield tempdir
+    finally:
+        dbfs_client.delete(tempdir, recursive=True)
+
+
+class ExtDbfsContextInjector(ExtContextInjector):
```

Comment:
its probably not the most consequential but I'm curious what would be the motivations to using this instead of `ExtEnvContextInjector`. Is it anything beyond just concerns over env var size limits? Does databricks call those limits out explicitly?

Reply:
In our meeting with Enigma, their guy said that they frequently run into size limits when passing their "context" (in our ontology, extras) over CLI, thought it would happen with env vars too, and explicitly suggested a DBFS mechanism.